Make -flarge-toc the default for PowerPC

dra27 opened the pull request:
Introduce `-fsmall-toc` in order to access the previous behaviour and document both options in the manual and ocamlopt manpage.

Addresses #9549

xavierleroy commented on man/ocamlopt.m:
```diff
@@ -728,6 +728,19 @@ The AMD64 code generator (64-bit versions of Intel Pentium and AMD
+greater than 64 kbytes.  This is the default since 4.11.
+.TP
+.B \-fsmall\-toc
+Enables the PowerPC small model allowing the TOC to be up to 64 kbytes.  Prior
```
Technically, it's 64 kbytes **per compilation unit**.  But I'm not sure we need to be that precise.

xavierleroy commented on man/ocamlopt.m:
```diff
@@ -728,6 +728,19 @@ The AMD64 code generator (64-bit versions of Intel Pentium and AMD
+.TP
+.B \-flarge\-toc
+Enables the PowerPC large model allowing the TOC (table of contents) to be
+greater than 64 kbytes.  This is the default since 4.11.
```
I would be tempted to say "arbitrarily large" because the limit is now 4 Gbytes.

xavierleroy reviewed (approved):
Looks good to me.  Two minor suggestions for the documentation, can be ignored if you don't like them.

dra27 committed:
Make -flarge-toc the default for PowerPC

Introduce -fsmall-toc in order to access the previous behaviour and
document both options in the manual and ocamlopt manpage.

dra27 commented:
Both changes sound good to me - done (manual and man page) and Changes updated. I'll merge once CI passes, thanks.

@Octachron - just confirming that this will go into 4.11. It's a change in behaviour, but it's not expected to qualify as a breaking change IIUC.

@kit-ty-kate - are the differential builds for 4.11 run on one of the ppc64 machines as well?

xavierleroy commented:
Travis and Appveyor seem stuck, but it doesn't matter since they don't test PPC.  CI precheck passes on PPC64LE.  I'll merge.

xavierleroy commented:
Cherry-picked to 4.11 branch: 03a9f5c81
